modules/oadp-gcp-wif-cloud-authentication.adoc

```diff
@@ -0,0 +1,94 @@
+
+With identity federation (WIF), you can use Identity and Access Management (IAM) to grant external identities IAM roles, including the ability to impersonate service accounts. This eliminates the maintenance and security risks associated with service account keys.
+
+WIF handles encrypting and decrypting certificates, extracting user attributes, and validation. WIF externalizes authentication, passing it over to Security Token Services (STS) and reduces the demands on individual developers. Authorization and controlling access to resources remain the responsibility of the application.
+
+[NOTE]
+====
+Google WIF cloud authentication is available for OADP 1.3.x and later.
+====
+
+If you do not use Google WIF cloud authentication, continue to xref:../../../backup_and_restore/application_backup_and_restore/installing/oadp-installing-operator.adoc#oadp-installing-operator-doc[Installing the OADP Operator].
+
+.Prerequisites
+
+* You must install a cluster in manual mode with link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.13/html-single/authentication_and_authorization/index#gcp-workload-identity-mode-installing[GCP Workload Identity configured].
+* You now have access to `ccoctl` CLI and access to the associated workload-identity-pool.
+
+.Procedure
+
+. Create an `oadp-credrequest` directory.
++
+[source,terminal]
+----
+$ mkdir -p oadp-credrequest
+----
++
+. Create a `CredentialsRequest`.yaml file as following:
++
+[source,yaml]
+----
+echo 'apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: oadp-operator-credentials
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: GCPProviderSpec
+    permissions:
+    - compute.disks.get
+    - compute.disks.create
+    - compute.disks.createSnapshot
+    - compute.snapshots.get
+    - compute.snapshots.create
+    - compute.snapshots.useReadOnly
+    - compute.snapshots.delete
+    - compute.zones.get
+    - storage.objects.create
+    - storage.objects.delete
+    - storage.objects.get
+    - storage.objects.list
+    - iam.serviceAccounts.signBlob
+    skipServiceCheck: true
+  secretRef:
+    name: cloud-credentials-gcp
+    namespace: <OPERATOR_INSTALL_NS>
+  serviceAccountNames:
+  - velero
+' > oadp-credrequest/credrequest.yaml
+----
++
+. Use `ccoctl` to create the `credrequest` pointing to the directory `oadp-credrequest`as following:
++
+[source,terminal]
+----
+ccoctl gcp create-service-accounts --name=<name> \
+    --project=<gcp-project-id> \
+    --credentials-requests-dir=oadp-credrequest \
+    --workload-identity-pool=<pool-id> \
+    --workload-identity-provider=<provider-id>
+----
++
+. The `manifests/openshift-adp-cloud-credentials-gcp-credentials.yaml` is now available to use in the following step.
```

Comment:
This is not a user step. If you need to call this out, you can still do that just without the `.`
```suggestion
The `manifests/openshift-adp-cloud-credentials-gcp-credentials.yaml` file is available to use in the following step.
```